scm/github.go

```diff
@@ -717,26 +601,130 @@ func (s *GithubSCM) CreateCourse(ctx context.Context, opt *NewCourseOptions) ([]
+		ID: opt.OrganizationID,
+	})
+	if err != nil {
+		return err
+	}
+	if _, err := s.client.Organizations.RemoveMember(ctx, org.Name, opt.User); err != nil {
+		return err
+	}
+	return s.DeleteRepository(ctx, &RepositoryOptions{ID: opt.RepositoryID})
+}
+
+// RevokeTeacherStatus removes user from teachers team, revokes owner status in the organization.
+func (s *GithubSCM) RevokeTeacherStatus(ctx context.Context, opt *UpdateEnrollmentOptions) error {
+	if _, err := s.client.Teams.RemoveTeamMembershipBySlug(ctx, opt.Organization, TeachersTeam, opt.User); err != nil {
+		return err
+	}
+	if _, _, err := s.client.Teams.AddTeamMembershipBySlug(ctx, opt.Organization, StudentsTeam, opt.User,
+		&github.TeamAddTeamMembershipOptions{Role: TeamMember}); err != nil {
+		return err
+	}
+	role := OrgMember
+	_, _, err := s.client.Organizations.EditOrgMembership(ctx, opt.User, opt.Organization, &github.Membership{Role: &role})
+	return err
+}
+
+// createStudentRepo creates {username}-labs repository and provides pull/push access to it for the given student.
+func (s *GithubSCM) createStudentRepo(ctx context.Context, org *qf.Organization, login string) (*Repository, error) {
+	// create repo, or return existing repo if it already exists
+	// if repo is found, it is safe to reuse it
 	repo, err := s.CreateRepository(ctx, &CreateRepositoryOptions{
 		Organization: org.Name,
-		Path:         qf.StudentRepoName(student),
-		Private:      private,
+		Path:         qf.StudentRepoName(login),
+		Private:      true,
 	})
 	if err != nil {
 		return nil, fmt.Errorf("failed to create repo: %w", err)
 	}
 
-	// Enable push and pull access to student repo
-	if err = s.UpdateRepoAccess(ctx, repo, student, RepoPush); err != nil {
-		return nil, fmt.Errorf("failed to enable push/pull access for repo: %w", err)
+	// add push access to student repo
+	opt := &github.RepositoryAddCollaboratorOptions{
+		Permission: RepoPush,
+	}
+	if _, _, err := s.client.Repositories.AddCollaborator(ctx, repo.Owner, repo.Path, login, opt); err != nil {
+		return nil, fmt.Errorf("failed to update repo push access: %w", err)
 	}
 	return repo, nil
 }
 
+func (s *GithubSCM) grantAccessToCourseRepos(ctx context.Context, org, login string) error {
+	commonRepos := []string{qf.InfoRepo, qf.AssignmentsRepo}
+
+	for _, repoType := range commonRepos {
+		opt := &github.RepositoryAddCollaboratorOptions{
+			Permission: RepoPull,
+		}
+		if _, _, err := s.client.Repositories.AddCollaborator(ctx, org, repoType, login, opt); err != nil {
+			return fmt.Errorf("updateReposAndTeams: failed to update repo access to repo %s for user %s: %w ", repoType, login, err)
+		}
+	}
+	return nil
+}
+
+// addUserToStudentsTeam adds user to the organization's "students" team.
+func (s *GithubSCM) addUserToStudentsTeam(ctx context.Context, org, login string) error {
+	_, _, err := s.client.Teams.AddTeamMembershipBySlug(ctx, org, StudentsTeam, login,
+		&github.TeamAddTeamMembershipOptions{Role: TeamMember})
```

Comment:
I would prefer that the options argument be prepared before the call and pass in, instead of on a new line.
```go
teamMember := &github.TeamAddTeamMembershipOptions{Role: TeamMember}
```

Reply:
The same applies to several other methods that pass a `TeamAddTeamMembershipOptions` argument ...